guides/keras_cv/random_walks_with_stable_diffusion.py

```diff
@@ -0,0 +1,247 @@
+Note that in order to maintain some stability between the resulting images we
+keep the diffusion noise constant between images.
+"""
+
+seed = 1337
+noise = tf.random.normal((512 // 8, 512 // 8, 4), seed=seed)
+
+images = model.generate_image(
+    interpolated_encodings,
+    batch_size=interpolation_steps,
+    diffusion_noise=noise,
+)
+
+"""
+Now that we've generated some interpolated images, let's take a look at them!
+
+Throughout this tutorial, we're going to export sequences of images as gifs so
+that they can be easily viewed with some temporal context. For sequences of
+images where the first and last images don't match conceptually, we rubber-band
+the gif.
+"""
+
+def export_as_gif(filename, images, frames_per_second=20, rubber_band=False):
+    if rubber_band:
+        images += images[2:-1][::-1]
+    images[0].save(filename, save_all=True, append_images=images[1:],
+                   duration=1000 // frames_per_second, loop=0)
+
+export_as_gif("doggo-and-fruit-5.gif", [Image.fromarray(img) for img in images],
+              frames_per_second=2, rubber_band=True)
+
+"""
+![Dog to Fruit 5](https://imgur.com/a/617Qjxb)
+
+The results may seem surprising. Generally, interpolating between prompts
+produces coherent looking images, and often demonstrate a progressive concept
+shift between the contents of the two prompts.
+
+To best visualize this, we should do a much more fine-grained interpolation,
+using hundreds of steps. In order to keep batch size small (so that we don't
+OOM our accelerators), this requires manually batching our interpolated
+encodings.
+"""
+
+interpolation_steps = 160
+batch_size = 16
+
+encoding_step = (encoding_2 - encoding_1) / interpolation_steps
+encoding_step = tf.squeeze(encoding_step)
+
+images = []
+for step_index in range(interpolation_steps // batch_size):
+    step_offsets = tf.linspace(batch_size*step_index,
+                               batch_size*(step_index+1) - 1, batch_size)
+    step_offsets = tf.tensordot(tf.cast(step_offsets, encoding_step.dtype),
+                                encoding_step, 0)
+    encodings = encoding_1 + tf.cast(step_offsets, encoding_1.dtype)
+    images += [Image.fromarray(img) for img in model.generate_image(
+        encodings,
+        batch_size=batch_size,
+        num_steps=25,
+        diffusion_noise=noise,
+    )]
+
+export_as_gif("doggo-and-fruit-160.gif", images, rubber_band=True)
+
+"""
+![Dog to Fruit 160](https://imgur.com/a/iNfRnH4)
+
+The resulting gif shows a much clearer and more coherent shift between the two
+prompts. Try out some prompts of your own and experiment!
+
+## A Random Walk Around a Text Prompt
+
+Our next experiment will be to go for a random walk around the latent manifold
+starting from a point produced by a particular prompt.
+"""
+
+walk_steps = 160
+batch_size = 16
+step_size = .05
+
+encoding = model.encode_text("A rubber duck swimming in a bowl of cereal")
```

Comment:
Should we just raw model.encoder.predict?  Or is this too complicated for that (as it relies on the embeddding too)

Reply:
IMO that would get a little ugly since this code would then have to deal with the pos_ids for the encoder as well as tensorification of the string. Maybe in the long run we should update the API to make model.encoder.encode or something visible, but in its current state I think this is the best option

Reply:
Seems reasonable!